File: NextTaskServer/app/schemas/message.py
from pydantic import BaseModel, Field
from typing import Optional, List, Any
from datetime import datetime

class MessageAuthor(BaseModel):
    id: int
    name: Optional[str] = None
    email: str
    avatar: Optional[str] = None

class MessageBase(BaseModel):
    id: int
    content: str
    is_edited: bool = False
    created_at: str
    updated_at: Optional[str] = None
    attachments: Optional[List[str]] = None
    reply_to_id: Optional[int] = None
    is_pinned: bool = False

class RepliedMessageRef(MessageBase):
    sender: Optional[MessageAuthor] = None

class Message(MessageBase):
    sender_id: int
    receiver_id: Optional[int] = None
    chat_id: Optional[int] = None
    is_read: int
    sender: Optional[MessageAuthor] = None
    receiver: Optional[MessageAuthor] = None
    replied_message: Optional[RepliedMessageRef] = None

class MessageCreate(BaseModel):
    receiver_id: Optional[int] = None
    chat_id: Optional[int] = None
    content: str
    attachments: Optional[List[str]] = None
    reply_to_id: Optional[int] = None

class MessageUpdate(BaseModel):
    content: Optional[str] = None
    attachments: Optional[List[str]] = None
    is_pinned: Optional[bool] = None

class ChatMemberSchema(BaseModel):
    id: int
    user_id: int
    chat_id: int
    joined_at: datetime
    user: Optional[MessageAuthor] = None

class ChatSchema(BaseModel):
    id: int
    name: Optional[str] = None
    is_group: bool = False
    created_at: datetime
    members: List[ChatMemberSchema] = []

class UnreadCountResponse(BaseModel):
    count: int

class RecentChatItem(BaseModel):
    id: str
    type: str  # "personal" | "group"
    user_id: Optional[int] = None
    chat_id: Optional[int] = None
    name: str
    avatar: Optional[str] = None
    last_activity_at: Optional[datetime] = None
    workspace_id: Optional[int] = None

class RecentChatsResponse(BaseModel):
    chats: List[RecentChatItem]
from pydantic import BaseModel
from typing import List, Optional

class ChatCreateRequest(BaseModel):
    name: str
    user_ids: List[int]

class ChatUpdateRequest(BaseModel):
    name: str

class ChatAddMembersRequest(BaseModel):
    user_ids: List[int]

File: NextTaskClient/src/shared/types/message.ts
// Типы для сообщений чата
export interface MessageAuthor {
	id: number;
	name?: string;
	email: string;
	avatar?: string;
}

export interface RepliedMessageRef {
	id: number;
	content: string;
	is_edited?: boolean;
	created_at: string;
	sender?: MessageAuthor;
}

export interface Message {
	id: number;
	sender_id: number;
	receiver_id?: number | null;
	chat_id?: number | null;
	content: string;
	is_read: number;
	is_edited?: boolean;
	is_pinned?: boolean;
	reply_to_id?: number | null;
	replied_message?: RepliedMessageRef | null;
	created_at: string;
	updated_at?: string | null;
	attachments?: string[];
	sender?: MessageAuthor;
	receiver?: MessageAuthor;
}

export interface MessageCreate {
	receiver_id?: number | null;
	chat_id?: number | null;
	content: string;
	attachments?: string[];
	reply_to_id?: number | null;
}

export interface MessageUpdate {
	content?: string;
	attachments?: string[];
	is_pinned?: boolean;
}

export interface ChatMember {
	id: number;
	user_id: number;
	chat_id: number;
	joined_at: string;
	user?: MessageAuthor;
}

export interface GroupChat {
	id: number;
	name?: string;
	is_group: boolean;
	created_at: string;
	members: ChatMember[];
}
